### PR TITLE
bump dependencies, use implementation, java 1.8 compat

### PR DESCRIPTION
**Pull request by TacoTheDank**
## General
- android studio 3.2.0 -> 3.2.1
- gradle 4.6 -> 4.10.3
- bump 28.0.2 -> 28.0.3 and add licenses in .travis.yml
- java 1.8 compatibility

## pftpd-pojo-lib/build.gradle
- replace compile with implementation
- commons-codec:commons-codec 1.10 -> 1.11
- ~~junit:junit 4.12 -> 4.13-beta-1~~

## primitiveFTPd/build.gradle
- ~~delete slf4j-android local library and use implementation~~
- ~~target android pie~~
- org.greenrobot:eventbus 3.0.0 -> 3.1.1
- eu.chainfire:libsuperuser 1.0.0.201704021214 -> 1.0.0.201811281328

### CODE REVIEW

**Comment by wolpi** [comment]
Thanks for taking care about this. But dependencies cannot be updated that easily. As you see the build fails.

These two have biggest issues:
> delete slf4j-android local library and use implementation

This app contains a copy and patches as it requires to change `LoggerFactory` at runtime. I could not find a way to do this with slf4j API.


> target android pie

This is currently not possible due to #157 

And I wonder why you would use a beta version of junit?

Would you like to create a PR with just the easy updates?

**Comment by TacoTheDank** [comment]
Hopefully this is better

File: pftpd-pojo-lib/build.gradle [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 }
 
 dependencies {
-    compile 'org.bouncycastle:bcpg-jdk15on:1.60'
-    compile 'org.bouncycastle:bcpkix-jdk15on:1.60'
+    implementation 'org.bouncycastle:bcpg-jdk15on:1.60'
+    implementation 'org.bouncycastle:bcpkix-jdk15on:1.60'
 
-    compile 'org.slf4j:slf4j-api:1.6.1'
+    implementation 'org.slf4j:slf4j-api:1.6.1' //1.7.25
```
Comment (wolpi) [inline pftpd-pojo-lib/build.gradle]: Remove the comment, please

File: pftpd-pojo-lib/build.gradle [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 
-    testCompile 'junit:junit:4.12'
-    testCompile 'commons-codec:commons-codec:1.10'
+    testImplementation 'junit:junit:4.13-beta-1'
```
Comment (wolpi) [inline pftpd-pojo-lib/build.gradle]: Don't use a beta version, please. Or do you have a good reason?

File: primitiveFTPd/build.gradle [hunk 3]
```diff
@@ -76,8 +69,8 @@ dependencies {
     implementation project(':pftpd-pojo-lib')
     implementation project(':slf4j-android-1.6.1-RC1')
     implementation 'org.apache.ftpserver:ftpserver-core:1.1.1'
-    implementation 'org.apache.sshd:sshd-core:0.14.0'
-    implementation 'org.greenrobot:eventbus:3.0.0'
+    implementation 'org.apache.sshd:sshd-core:0.14.0' //2.1.0
```
Comment (wolpi) [inline primitiveFTPd/build.gradle]: Remove the comment, please.

File: slf4j-android-1.6.1-RC1/build.gradle [hunk 4]
```diff
@@ -17,9 +15,9 @@ jar {
 }
 
 repositories {
-    jcenter();
+    jcenter()
 }
 
 dependencies {
-    compileOnly 'com.google.android:android:2.1.2'
+    compileOnly 'com.google.android:android:2.1.2' //4.1.1.4
```
Comment (wolpi) [inline slf4j-android-1.6.1-RC1/build.gradle]: Remove the comment, please.

**Comment by wolpi** [comment]
it is failing in travis again :cry:

**Comment by TacoTheDank** [comment]
🤔 
```
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':primitiveFTPd:processDebugResources'.
> Android resource linking failed
  warn: removing resource org.primftpd:string/prefCategoryTitle without required default value.
  warn: removing resource org.primftpd:string/prefForegroundService without required default value.
  warn: removing resource org.primftpd:string/prefSummaryForegroundService without required default value.
  /home/travis/build/wolpi/prim-ftpd/primitiveFTPd/res/mipmap-anydpi/ic_launcher.xml: error: <adaptive-icon> elements require a sdk version of at least 26.
  error: failed linking file resources.
```

File: pftpd-pojo-lib/build.gradle [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
 
 repositories {
     jcenter()
 }
 
 dependencies {
-    compile 'org.bouncycastle:bcpg-jdk15on:1.60'
-    compile 'org.bouncycastle:bcpkix-jdk15on:1.60'
+    implementation 'org.bouncycastle:bcpg-jdk15on:1.60'
+    implementation 'org.bouncycastle:bcpkix-jdk15on:1.60'
```
Comment (wolpi) [inline pftpd-pojo-lib/build.gradle]: Building locally on my box fails :cry: 
Compiler error:
```
org/primftpd/services/PubKeyAuthenticator.java:42: error: package org.bouncycastle.jce.provider does not exist
            if (serverKey instanceof org.bouncycastle.jce.provider.JCEECPublicKey)
```
Are `implementation` jars visible to dependent projects?

**Comment by wolpi** [comment]
see new file comment please

**Comment by TacoTheDank** [comment]
I'll close this because it's old, and I'm cleaning up my repos.

**Pull request closed without merging**